sdlib/d/gc/tcache.d

```diff
@@ -83,20 +121,20 @@ public:
 			return alloc(size, containsPointers);
 		}
 
-		auto copySize = size;
 		auto pd = getPageDescriptor(ptr);
+		CapacityInfo info;
+
 		auto samePointerness = containsPointers == pd.containsPointers;
 
 		if (pd.isSlab()) {
 			auto newSizeClass = getSizeClass(size);
 			auto oldSizeClass = pd.sizeClass;
 			if (samePointerness && newSizeClass == oldSizeClass) {
+				setSmallUsedCapacity(pd, ptr, size);
 				return ptr;
 			}
 
-			if (newSizeClass > oldSizeClass) {
-				copySize = getSizeFromClass(oldSizeClass);
-			}
+			info = CapacityInfo(pd, ptr);
```

Comment:
I don't think this is worth it. All the extra work to avoid copying a few bytes.